sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOReadTest.java

```diff
@@ -906,4 +1093,227 @@ public void testCoderInference() {
         KvCoder.of(ByteStringCoder.of(), ProtoCoder.of(Mutation.class)),
         BigQueryIO.read(parseFn).inferCoder(CoderRegistry.createDefault()));
   }
+
+  @Test
+  public void testRowProtoCoderInference() {
+    SerializableFunction<SchemaAndRowProto, KV<ByteString, Mutation>> parseFn =
+        new SerializableFunction<SchemaAndRowProto, KV<ByteString, Mutation>>() {
+          @Override
+          public KV<ByteString, Mutation> apply(SchemaAndRowProto input) {
+            return null;
+          }
+        };
+
+    assertEquals(
+        KvCoder.of(ByteStringCoder.of(), ProtoCoder.of(Mutation.class)),
+        BigQueryIO.readViaRowProto(parseFn).inferCoder(CoderRegistry.createDefault()));
+  }
+
+  private CreateSessionRequest.Builder getDefaultCreateSessionRequestBuilder() {
```

Comment:
This is a lot of new code. What about moving the tests of the new codepath into a separate file? We already split BigQueryIOTest once into read + write, seems like there's enough new test code for the new Read that it deserves a separate file also.

Reply:
I can split this into a separate test file.

Reply:
Done.